### DISCUSSION TITLE
could not open the consult ripgrep location

### DISCUSSION THREAD

**Discussion by zhuoqiang** [minad/consult#1087] [Q&A] [answered]
here's the config:

```list
(use-package consult
  :bind (("C-r" . consult-ripgrep)))
```

I would like to 

1. `C-r` and search some text
2. `C-n` / `C-p` to view/select the search locations
3. `C-m` to jump/open the location of current selected location

however, when I hit enter / `C-m`, it just open the first location but not the location highlighted.

I have to use `tab` to select in another buffer

besides, I would like to open the search result in a normal grep-like buffer so that I could move and search in that buffer freely and hit enter to go to specific location. I tried `C-o` in the result buffer but did not work.

I remember the above works in some old configuration, but failed in latest with emacs 29, mac.

**Comment by zhuoqiang** [comment, marked as answer]
I may found the root cause: the `icomplete-vertical-mode` is incompatible with consult. disable `icomplate-vertical-mode` resovle the issue.

```lisp
  ;; incompatible with consult
  ;; (icomplete-vertical-mode 1)
```

**Reply by minad** [reply]
No, Icomplete shouldn't be incompatible. I haven't tested it in a while, but there haven't been any significant changes in icomplete-mode recently. You've probably not configured Icomplete correctly, such that it matches your expectations.

**Reply by oantolin** [reply]
I think that in icomplete the default binding for RET (or equivalently C-m) only accepts a candidate when it is the unique match, i.e., under the hood it does minibuffer-complete-and-exit. You want C-j instead, which forces completion to the current candidate even if there are other matches. (Helpfully, but also confusingly, in fido-mode RET *does* force completion!)

In other words, @zhuoqiang, in icomplete-mode or icomplete-vertical-mode you should use C-j to select a consult candidate, not C-m.

**Reply by zhuoqiang** [reply]
@oantolin tried that, with `(icomplete-vertical-mode 1)` I could not even navigate the condidates in minibuffer using `C-n` and `C-p`

- Box: Apple M1 Pro
- OS: macOS 14.6.1
- Emacs: GNU Emacs 29.4 (build 1, aarch64-apple-darwin21.6.0, NS appkit-2113.60 Version 12.6.6 (Build 21G646)) of 2024-08-03

**Reply by zhuoqiang** [reply]
BTW, any way to do immediatet action while switching buffers. I would like to kill buffer using `C-k` while navigating the candidate in minibuffer.

right now I have to `C-a` to bring up a *Embark Action* and then `K` to kill
